src/internal/drawImage.js

```diff
@@ -6,9 +17,169 @@
  * @returns {void}
  * @memberof Internal
  */
-export default function (enabledElement, invalidated = false) {
-  enabledElement.needsRedraw = true;
+export function drawImage (enabledElement, invalidated = false) {
+  registerVisibilityChange();
+
+  if (!enabledElement || !enabledElement.canvas || (!enabledElement.image && enabledElement.layers.length < 1)) {
+    return;
+  }
+  
+  const element = enabledElement.element;
+  
   if (invalidated) {
     enabledElement.invalid = true;
   }
+
+  if (!map.has(element)) {
+    map.set(element, { 
+      needsRedraw: false, 
+      callbacks: new Set()
+    });
+  }
+
+  map.get(element).needsRedraw = true;
+
+  rafStopTime = performance.now() + 100;
+  if (!rafId) {
+    rafId = requestAnimationFrame(step);
+  }
+}
+
+function step () {
+  map.forEach((entry, element) => {
+    if (!element || !entry || !entry.needsRedraw) {
+      return;
+    }
+    
+    const hasCallbacks = entry.callbacks.size > 0;
+    
+    entry.needsRedraw = hasCallbacks;
+
+    let enabledElement;
+
+    try { 
+      enabledElement = getEnabledElement(element); 
+    } catch (err) { 
+      return; 
+    }
+    
+    const eventDetails = {
+      enabledElement,
+      timestamp: Date.now()
+    };
+    
+    triggerEvent(element, EVENTS.PRE_RENDER, eventDetails);
+    
+    drawImageSync(enabledElement, enabledElement.invalid);
+
+    if (hasCallbacks) {
+      entry.callbacks.forEach((rafcb) => {
+        try {
+          rafcb(enabledElement);
+        } catch (err) {
+          // swallow
+        }
+      });
+      rafStopTime = performance.now() + 100; 
+    }
+  });
+  
+  if (rafStopTime && performance.now() < rafStopTime) {
```

Comment:
I don't fully understand why you keep it running for 100ms once triggered.
Also, could you replace the two `100` occurrences by a more informative constant name.

Reply:
The whole idea is to keep bouncing the call to cancelAnimationFrame forward so that if no draw requests were received within a 100ms period RAF will be cancelled (and restarted on a new request)

Reply:
(added constant rafStopMaxTime)